algorithms/src/snark/marlin/data_structures/proof.rs

```diff
@@ -296,13 +346,14 @@ impl<E: PairingEngine> CanonicalDeserialize for Proof<E> {
         compress: Compress,
         validate: Validate,
     ) -> Result<Self, SerializationError> {
-        let batch_size = CanonicalDeserialize::deserialize_with_mode(&mut reader, compress, validate)?;
+        let batch_sizes: Vec<u64> = CanonicalDeserialize::deserialize_with_mode(&mut reader, compress, validate)?;
+        let batch_sizes: Vec<usize> = batch_sizes.into_iter().map(|x| x as usize).collect();
```

Comment:
these casts will truncate on 32bit arch, if we're targeting any

Reply:
@Pratyush I'm thinking again it would be a good idea to change these usize's throughout the codebase to u64 because:
- (1) we are not checking whether instances to be proven are unique. So an attacker could cheaply create 2**32 copies of an instance in a single proof to a 32-bit verifier, and 1+2**32 copies of an instance in a single proof to a 64-bit verifier, and somehow cause a fork if some part of the consensus algorithm or our code assumes a proof corresponds 1:1 to a state transition.
- (2) even if there is not a vulnerability, developers on snarkVM will keep spending time on this and question this design decision till the end of time

Reply:
That sounds good to me! Changing to u64 is completely fine. We can even keep it at, say, u16; I don't expect people to have batches larger than 2^16